From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Mon, 19 Aug 2024 19:30:52 -0500
Subject: [PATCH] Revert "build system: ccenv_set_cc_underscore(): rewrite
 against strict posix utils."

This reverts commit d6df7b30e6b9a59388d7adfd671cc83b25342ea7.
---
 sofort/ccenv/ccenv.sh | 22 ++++++----------------
 1 file changed, 6 insertions(+), 16 deletions(-)

diff --git a/sofort/ccenv/ccenv.sh b/sofort/ccenv/ccenv.sh
index d975c6b..1c9d80c 100644
--- a/sofort/ccenv/ccenv.sh
+++ b/sofort/ccenv/ccenv.sh
@@ -806,26 +806,16 @@ ccenv_set_cc_underscore()
 	ccenv_fn_name='ZmYaXyWbVe_UuTnSdReQrPsOcNoNrLe'
 	ccenv_fn_code='int %s(void){return 0;}'
 
-	ccenv_tmpname='ccenv/c3RyaWN0X21vZGUK.c'
-
-	printf "$ccenv_fn_code" $ccenv_fn_name \
-		> "$ccenv_tmpname"
-
-	$ccenv_cc -c "$ccenv_tmpname" -o a.out \
-		> /dev/null 2>&3
-
-	if "$ccenv_nm" a.out | grep          \
-			-e "^_$ccenv_fn_name" \
-			-e " _$ccenv_fn_name"  \
-			> /dev/null; then
+	if printf "$ccenv_fn_code" $ccenv_fn_name  \
+			| $ccenv_cc -xc - -S -o -  \
+				2>&3               \
+			| grep "^_$ccenv_fn_name:" \
+				> /dev/null; then
 		ccenv_cc_underscore='_'
 		ccenv_attr_epilog 'yes'
-	else
-		ccenv_attr_epilog 'no'
 	fi
 
-	rm "$ccenv_tmpname"
-	rm a.out
+	ccenv_attr_epilog 'no'
 
 	return 0
 }
